de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/InstantOutgoingProjectNegotiation.java

```diff
@@ -0,0 +1,273 @@
+package de.fu_berlin.inf.dpp.negotiation;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.LinkedBlockingDeque;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.log4j.Logger;
+import org.jivesoftware.smackx.filetransfer.FileTransfer;
+import org.jivesoftware.smackx.filetransfer.OutgoingFileTransfer;
+
+import de.fu_berlin.inf.dpp.activities.SPath;
+import de.fu_berlin.inf.dpp.editor.AbstractSharedEditorListener;
+import de.fu_berlin.inf.dpp.editor.IEditorManager;
+import de.fu_berlin.inf.dpp.editor.ISharedEditorListener;
+import de.fu_berlin.inf.dpp.editor.remote.UserEditorStateManager;
+import de.fu_berlin.inf.dpp.exceptions.LocalCancellationException;
+import de.fu_berlin.inf.dpp.exceptions.SarosCancellationException;
+import de.fu_berlin.inf.dpp.filesystem.IChecksumCache;
+import de.fu_berlin.inf.dpp.filesystem.IProject;
+import de.fu_berlin.inf.dpp.filesystem.IWorkspace;
+import de.fu_berlin.inf.dpp.monitoring.IProgressMonitor;
+import de.fu_berlin.inf.dpp.negotiation.NegotiationTools.CancelOption;
+import de.fu_berlin.inf.dpp.negotiation.stream.OutgoingStreamProtocol;
+import de.fu_berlin.inf.dpp.net.IReceiver;
+import de.fu_berlin.inf.dpp.net.ITransmitter;
+import de.fu_berlin.inf.dpp.net.xmpp.JID;
+import de.fu_berlin.inf.dpp.net.xmpp.XMPPConnectionService;
+import de.fu_berlin.inf.dpp.session.ISarosSession;
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+import de.fu_berlin.inf.dpp.session.User;
+import de.fu_berlin.inf.dpp.synchronize.StartHandle;
+
+/**
+ * Share Projects to display them instant on client side using a stream based
+ * solution.
+ */
+public class InstantOutgoingProjectNegotiation extends
+    AbstractOutgoingProjectNegotiation {
+
+    private static final Logger log = Logger
+        .getLogger(InstantOutgoingProjectNegotiation.class);
+
+    /** used as LIFO queue **/
+    private final Deque<SPath> openedFiles = new LinkedBlockingDeque<SPath>();
+
+    private Set<SPath> transferList;
+    private Set<SPath> transmittedFiles;
+
+    /** receive open editors to prioritize these files **/
+    private final ISharedEditorListener listener = new AbstractSharedEditorListener() {
+        @Override
+        public void editorActivated(User user, SPath file) {
+            fileOpened(file);
+        }
+    };
+
+    private List<StartHandle> stoppedUsers = null;
+    private User remoteUser = null;
+
+    public InstantOutgoingProjectNegotiation(final JID peer, //
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Constructor has 10 parameters, which is greater than 7 authorized. ([squid:S00107](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS00107))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=128&issues=AWSS35X1sk71k7lwr5eQ&open=AWSS35X1sk71k7lwr5eQ)

[ISSUE_KEY]: AWSS35X1sk71k7lwr5eQ